cql3/statements/modification_statement.cc

```diff
@@ -529,6 +529,19 @@ modification_statement::validate(service::storage_proxy&, const service::client_
     if (is_view()) {
         throw exceptions::invalid_request_exception("Cannot directly modify a materialized view");
     }
+
+    if (attrs->is_timestamp_set()) {
+        static constexpr int64_t MAX_DIFFERENCE = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::days(3)).count();
+        auto now = std::chrono::duration_cast<std::chrono::microseconds>(db_clock::now().time_since_epoch()).count();
+        cql3::query_options query_options(std::vector<cql3::raw_value>{cql3::raw_value::make_unset_value()});
+
```

Comment:
This looks like a wasteful way to get a timestamp, including an allocation for a vector.

Reply:
validate() runs on prepare time, while USING TIMESTAMP can have a value that is only available during execution time (`USING TIMESTAMP :my_variable`). Not sure how this works.

Reply:
Fair point, moved the checks to right before execution